src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/JsonGetFieldAsText.php

```diff
@@ -16,7 +16,7 @@ class JsonGetFieldAsText extends BaseFunction
 {
     protected function customiseFunction(): void
     {
-        $this->setFunctionPrototype('(%s ->> %s)');
+        $this->setFunctionPrototype('(to_json(%s) ->> %s)');
```

Comment:
This change introduces overhead in the query execution plan of PostgreSql. It is widely expected, that in the majority of the cases, the data is already in json/jsonb format, which makes the invocation of `to_json` unnecessary.

Imo, the use case you are trying to solve here shall be achieved through the introduction of a dedicated function, that accepts `text` data, transforms it to `json/jsonb` and only after that extracts a field from it.